### DISCUSSION TITLE
en/blog/nix-home-manager-flake-setup/

### DISCUSSION THREAD

**Discussion by giscus** [rexk/rexk.github.io#7] [General]
# en/blog/nix-home-manager-flake-setup/

Getting started with nix and home manager

https://rexk.github.io/en/blog/nix-home-manager-flake-setup/

**Comment by aimestereo** [comment]
Thanks for summing many things up in nice way, it's a first example that actually worked for me with some tweaks.
I have a few questions, though:
* how do you manage files? All of it is stored in git repository, but all scripts target `~/.config` folder which is not explicitly filled. i.e. I use `stow` to copy all my dotfiles/scripts from repo to ~/.config folder as a prepare step and set `XDG_CONFIG_HOME` to point to my repo instead of ~/.config folder more about it in next point.
* how targeting ~/.config folder in your scripts even works? or more simple example, how `user.nix` even works?, for me it works only if I generate and add it to repo and then `hms` script targets flake stored in this git repo, but it will beat the whole purpose of customization for different machines, it's due to https://github.com/NixOS/nix/issues/7107 NIX flakes ignore files that are not added to git.
